src/routerd_lib/handlers/proxy.cpp

```diff
@@ -45,88 +46,224 @@ namespace NAC {
 
                         for (const auto& part : response->Parts()) {
                             std::string partName;
                             NStringUtils::Strip(part.ContentDispositionParams().at("filename"), partName, 2, "\"'");
 
                             if (partName == service.Name) {
-                                haveDirectResponse = true;
+                                serviceReplyProcessed = true;
                             }
 
                             ProcessServiceResponse(request, response, partName, &part, /* contentDispositionFormData = */false);
                         }
-
-                        if (!haveDirectResponse) {
-                            ServiceReplied(request, service.Name);
-                        }
-
                     } else {
-                        ProcessServiceResponse(request, response, service.Name, response.get());
+                        if (!service.SaveAs.empty()) {
+                            ProcessServiceResponse(request, response, service.SaveAs, response.get());
+                        } else {
+                            ProcessServiceResponse(request, response, service.Name, response.get());
+                            serviceReplyProcessed = true;
+                        }
+                    }
+                    if (!serviceReplyProcessed) {
+                        ServiceReplied(request, service.Name);
                     }
 
-                    Iter(request, args);
+                    Iter(request, args); // recursion depth is limited by graph size, which is small.
                 });
 
-                if (!rv) {
+                if (!rv) { // could not connect
                     failedServices.push_back(service.Name);
                     continue;
                 }
 
-                auto msg = request->OutgoingRequest(service.Path, args);
-                msg.Memorize(request);
-
                 request->NewRequest(service.Name);
-                rv->PushWriteQueueData(msg);
+
+#ifdef AC_DEBUG_ROUTERD_PROXY
+                printOutgoingRequest(request);
+#endif
+                // schedule payload to be sent to connected service
+                if (!service.SendRawOutputOf.empty()) {
+                    auto&& outgoingRequest = request->GetOutGoingRequest();
+                    auto matchingPart = outgoingRequest.PartByName(service.SendRawOutputOf);
+                    if (matchingPart) {
+#ifdef AC_DEBUG_ROUTERD_PROXY
+                        std::cerr << "to service " << service.Name
+                                  << " will send_raw_output_of " << service.SendRawOutputOf << std::endl;
+#endif
+                        rv->PushWriteQueueData(matchingPart->GetBody());
+                    } else { // should not happen: we've checked it above
+                        request->Send500();
+#ifdef AC_DEBUG_ROUTERD_PROXY
+                        std::cerr << "raw output part not found, issuing 500" << std::endl;
```

Comment:
This message seems important, let's print it regardless of AC_DEBUG_ROUTERD_PROXY.